src/Init/Data/List/Basic.lean

```diff
@@ -456,11 +456,22 @@ def isPrefixOf [BEq α] : List α → List α → Bool
   | _,     []    => false
   | a::as, b::bs => a == b && isPrefixOf as bs
 
+/-- `getSuffix? l₁ l₂` returns `some t` when `l₂ == l₁ ++ t`. -/
+def getSuffix? [BEq α] : List α → List α → Option (List α)
```

Comment:
`isSuffix?` might be a little more consistent with the rest of the library (e.g. `Syntax.isNatLit?`).

Reply:
on the other hand, this would mean that `isPrefix` and `isPrefix?` would be doing different things which might be confusing. I was hoping there would be a fn in mathlib3 I could steal the name from.

Reply:
@digama0 what should it be called?

Reply:
What do you mean by them doing different things? We do have a naming convention that `?` means Option-returning, and it is true that `(l1.isPrefix? l2).isSome == l1.isPrefix l2` so it seems okay to me.

...Oh, are prefix and suffix switched in this definition? I would recommend using `l1.isPrefix? l2` for the function that returns t such that `l2 ++ t == l1`, and similarly for `isSuffix?`.